File: app/src/main/java/com/ivanovsky/passnotes/data/repository/RemoteFileRepository.java
package com.ivanovsky.passnotes.data.repository;

import com.ivanovsky.passnotes.data.entity.RemoteFile;
import com.ivanovsky.passnotes.data.repository.db.dao.RemoteFileDao;
import com.ivanovsky.passnotes.data.repository.file.FSType;

import java.util.List;

public class RemoteFileRepository {

	private final RemoteFileDao dao;

	public RemoteFileRepository(RemoteFileDao dao) {
		this.dao = dao;
	}

	public List<RemoteFile> getAllByFsType(FSType fsType) {
		return dao.getAll(fsType.getValue());
	}

	public RemoteFile findByUidAndFsType(String uid, FSType fsType) {
		return dao.findByUidAndFsType(uid, fsType.getValue());
	}

	public RemoteFile findByRemotePathAndFsType(String remotePath, FSType fsType) {
		return dao.findByRemotePathAndFsType(remotePath, fsType.getValue());
	}

	public void insert(RemoteFile link) {
		long id = dao.insert(link);
		link.setId(id);
	}

	public void update(RemoteFile link) {
		dao.update(link);
	}

	public void delete(int id) {
		dao.delete(id);
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/data/repository/RemoteFileRepository.kt
package com.ivanovsky.passnotes.data.repository

import com.ivanovsky.passnotes.data.entity.FSAuthority
import com.ivanovsky.passnotes.data.entity.RemoteFile
import com.ivanovsky.passnotes.data.repository.db.dao.RemoteFileDao

class RemoteFileRepository(
    private val dao: RemoteFileDao
) {

    fun getAll(fsAuthority: FSAuthority): List<RemoteFile> {
        return dao.all
            .filter { file -> file.fsAuthority == fsAuthority }
    }

    fun findByUid(uid: String, fsAuthority: FSAuthority): RemoteFile? {
        return dao.all
            .firstOrNull { file -> file.uid == uid && file.fsAuthority == fsAuthority }
    }

    fun findByRemotePath(remotePath: String, fsAuthority: FSAuthority): RemoteFile? {
        return dao.all
            .firstOrNull { file ->
                file.remotePath == remotePath && file.fsAuthority == fsAuthority
            }
    }

    fun insert(file: RemoteFile) {
        val id = dao.insert(file)
        file.id = id
    }

    fun update(file: RemoteFile) {
        dao.update(file)
    }

    fun delete(id: Int) {
        dao.delete(id.toLong())
    }
}